src/GafferSceneUI/SetMembershipInspector.cpp

```diff
@@ -216,14 +216,40 @@ IECore::ConstObjectPtr SetMembershipInspector::value( const GafferScene::SceneAl
 	return matchResult & IECore::PathMatcher::Result::ExactMatch ? new BoolData( true ) : nullptr;
 }
 
-IECore::ConstObjectPtr SetMembershipInspector::fallbackValue( const GafferScene::SceneAlgo::History *history ) const
+IECore::ConstObjectPtr SetMembershipInspector::fallbackValue( const GafferScene::SceneAlgo::History *history, std::string &description ) const
 {
 	const auto &path = history->context->get<ScenePlug::ScenePath>( ScenePlug::scenePathContextName );
 	ConstPathMatcherDataPtr setMembers = history->scene->set( m_setName );
 
 	auto matchResult = (PathMatcher::Result)setMembers->readable().match( path );
 
-	return new BoolData( matchResult & IECore::PathMatcher::Result::AncestorMatch );
+	const bool ancestorMatch = matchResult & IECore::PathMatcher::Result::AncestorMatch;
+	if( ancestorMatch )
+	{
+		ScenePlug::ScenePath currentPath( history->context->get<ScenePlug::ScenePath>( ScenePlug::scenePathContextName ) );
+		if( currentPath.size() > 1 )
```

Comment:
If the path has size 1 (`/sphere` say), then we're still doing an evaluation at the current location. I think this should just be `if( currentPath.size() )`? And maybe we could limit the additional nesting by doing `if( ancestorMatch && path.size() )` at L227 instead?

Reply:
Switched to `if( ancestorMatch && path.size() )` at L227 and squashed to d1c5ac3.